crates/oxc_linter/src/rules/jsx_a11y/no_redundant_roles.rs

```diff
@@ -0,0 +1,135 @@
+use crate::{context::LintContext, rule::Rule, utils::has_jsx_prop_lowercase, AstNode};
+use oxc_ast::{
+    ast::{JSXAttributeItem, JSXAttributeValue, JSXElementName},
+    AstKind,
+};
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::{self, Error},
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+use phf::{phf_map, phf_set};
+use std::collections::HashMap;
+
+#[derive(Debug, Error, Diagnostic)]
+#[error(
+    "eslint-plugin-jsx-a11y(no-redundant-roles): The element `{element}` has an implicit role of `{role}`. Defining this explicitly is redundant and should be avoided."
+)]
+#[diagnostic(
+    severity(warning),
+    help("Remove the redundant role `{role}` from the element `{element}`.")
+)]
+struct NoRedundantRolesDiagnostic {
+    #[label]
+    pub span: Span,
+    pub element: String,
+    pub role: String,
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct NoRedundantRoles;
+
+declare_oxc_lint!(
+    /// ### What it does
+    /// Enforces that the explicit role property is not the same as implicit/default role property on element.
+    ///
+    /// ### Why is this bad?
+    /// Redundant roles can lead to confusion and verbosity in the codebase.
+    ///
+    /// ### Example
+    /// ```javascript
+    /// // Bad
+    /// <nav role="navigation" />
+    ///
+    /// // Good
+    /// <nav />
+    /// ```
+    NoRedundantRoles,
+    correctness
+);
+
+static DEFAULT_ROLE_EXCEPTIONS: phf::Map<&'static str, phf::Set<&'static str>> = phf_map! {
+    "nav" => phf_set!{"navigation"},
+    "button" => phf_set!{"button"},
+    "body" => phf_set!{"document"},
+};
+
+static EMPTY_SET: phf::Set<&'static str> = phf_set! {};
+
+impl Rule for NoRedundantRoles {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        let components: HashMap<String, String> = ctx
+            .settings()
+            .jsx_a11y
+            .components
+            .iter()
+            .map(|(k, v)| (k.to_string(), v.clone()))
+            .collect();
+
+        if let AstKind::JSXOpeningElement(jsx_el) = node.kind() {
+            let element = match &jsx_el.name {
+                JSXElementName::Identifier(identifier) => identifier.name.to_string(),
+                JSXElementName::NamespacedName(namespaced_name) => {
+                    namespaced_name.property.name.to_string()
+                }
+                JSXElementName::MemberExpression(_) => return,
+            };
+
+            let component = components.get(&element).unwrap_or(&element);
```

Comment:
i think you can use `https://github.com/oxc-project/oxc/blob/64310fa9b110924f4e33c6fbb5e7676c0de718fa/crates/oxc_linter/src/utils/react.rs#L180-L201` instead?

Reply:
"Thank you! !! I see, there was no need to write it myself...
I made the correction at 8e14a669c3d9f3eeb121f85d6da162f70d414b7d!!"

Reply:
thanks!